cmd/service_linux.go

```diff
@@ -0,0 +1,340 @@
+		}
+
+		configExists := common.DirectoryExists(configPath)
+		if !configExists {
+			return fmt.Errorf("error, the configfile path provided does not exist") // TODO: add useage output upon failure with input
+		}
+
+		err = modifySericeFile(mountPath, dir)
+		if err != nil {
+			return fmt.Errorf("error when attempting to write defaults into service file: [%s]", err.Error())
+		}
+
+		err = modifySericeFile(configPath, dir)
+		if err != nil {
+			return fmt.Errorf("error when attempting to write defaults into service file: [%s]", err.Error())
+		}
+
+		// 2. retrieve the newUser account from cloudfuse.service file and create it if it doesn't exist
+
+		serviceUser := serviceData["User"]
+		setUser(serviceUser)
+
+		// 3. copy the cloudfuse.service file to /etc/systemd/system
+		copyFileCmd := exec.Command("sudo", "cp", "./setup/cloudfuse.service", "/etc/systemd/system")
+		err = copyFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to copy cloudfuse.service file to /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		// 4. run systemctl daemon-reload
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		// 5. Enable the service to start at system boot
+		systemctlEnableCmd := exec.Command("sudo", "systemctl", "enable", "cloudfuse.service")
+		err = systemctlEnableCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+var uninstallCmd = &cobra.Command{
+	Use:               "uninstall",
+	Short:             "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	Long:              "Uninstall the startup process for Cloudfuse. Requires running as admin.",
+	SuggestFor:        []string{"uninst", "uninstal"},
+	Example:           "cloudfuse service uninstall",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		// 1. find and remove cloudfuse.service from /etc/systemd/system and run systemctl daemon-reload
+		removeFileCmd := exec.Command("sudo", "rm", "/etc/systemd/system/cloudfuse.service")
+		err := removeFileCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to delete cloudfuse.service file from /etc/systemd/system due to following error: [%s]", err.Error())
+		}
+
+		systemctlDaemonReloadCmd := exec.Command("sudo", "systemctl", "daemon-reload")
+		err = systemctlDaemonReloadCmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func collectServiceData(serviceFilePath string) (map[string]string, error) {
```

Comment:
I don't know, but I feel like these helper functions could be a lot shorter, and more readable. I feel like these details should be abstracted away by a standard library function that does most or all of this for us...